### ISSUE TITLE
Adding ImageCollection objects does not propogate filter_codes attribute

### ISSUE THREAD

**Issue by tHarvey303** [flaresimulations/synthesizer#555]
I don't know if this issue is already known, but if you have two `ImageCollection` objects and you add them (e.g. a disk and bulge model for a galaxy), the `__add__` overload does not propogate the `filter_codes` attribute into the `composite_img` object . The `filter_codes` attribute is set to an empty list, leading to an error in the matplotlib figure setup when you call `self.plot_images()` because it tries to make a `GridSpec` with 0 columns (error below). It is a trivial fix around line 395 of `image_collection.py` - simply adding something like `composite_img.filter_codes.append(f)` , to the for loop makes it work, although a slightly more clever fix would ensure that the filter_codes list stays in a sensible order - currently it is jumbled by the use of a set, which is unordered. 

I'm using the current 0.4.0 beta version, but I think this behaviour has existed for a few months.  

Error:

```
['JWST/NIRCam.F090W', 'JWST/NIRCam.F115W', 'JWST/NIRCam.F150W', 'JWST/NIRCam.F200W', 'JWST/NIRCam.F277W', 'JWST/NIRCam.F356W', 'JWST/NIRCam.F444W']
['JWST/NIRCam.F090W', 'JWST/NIRCam.F115W', 'JWST/NIRCam.F150W', 'JWST/NIRCam.F200W', 'JWST/NIRCam.F277W', 'JWST/NIRCam.F356W', 'JWST/NIRCam.F444W']
[]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/home/tharvey/work/synthesizer/scripts/multi_component_galaxy.ipynb Cell 12 line 9
     [90](vscode-notebook-cell://wsl%2Bubuntu/home/tharvey/work/synthesizer/scripts/multi_component_galaxy.ipynb#X43sdnNjb2RlLXJlbW90ZQ%3D%3D?line=89) print(combined_image.filter_codes)
     [95](vscode-notebook-cell://wsl%2Bubuntu/home/tharvey/work/synthesizer/scripts/multi_component_galaxy.ipynb#X43sdnNjb2RlLXJlbW90ZQ%3D%3D?line=94) #combined_image.filter_codes = bulge_img.filter_codes
     [96](vscode-notebook-cell://wsl%2Bubuntu/home/tharvey/work/synthesizer/scripts/multi_component_galaxy.ipynb#X43sdnNjb2RlLXJlbW90ZQ%3D%3D?line=95) # Plot the images - doesn't work because combined_image.filter_codes is an empty list. 
---> [97](vscode-notebook-cell://wsl%2Bubuntu/home/tharvey/work/synthesizer/scripts/multi_component_galaxy.ipynb#X43sdnNjb2RlLXJlbW90ZQ%3D%3D?line=96) combined_image.plot_images()

File [~/software/synthesizer/src/synthesizer/imaging/image_collection.py:756](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:756), in ImageCollection.plot_images(self, show, vmin, vmax, scaling_func, cmap)
    [751](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:751) fig = plt.figure(
    [752](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:752)     figsize=(4 * 3.5, int(np.ceil(len(self.filter_codes) / 4)) * 3.5)
    [753](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:753) )
    [755](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:755) # Create a gridspec grid
--> [756](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:756) gs = gridspec.GridSpec(
    [757](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:757)     int(np.ceil(len(self.filter_codes) / 4)), 4, hspace=0.0, wspace=0.0
    [758](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:758) )
    [760](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:760) # Loop over filters making each image
    [761](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:761) for ind, f in enumerate(self.filter_codes):
    [762](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/software/synthesizer/src/synthesizer/imaging/image_collection.py:762)     # Get the image

File [~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:378](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:378), in GridSpec.__init__(self, nrows, ncols, figure, left, bottom, right, top, wspace, hspace, width_ratios, height_ratios)
    [375](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:375) self.hspace = hspace
    [376](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:376) self.figure = figure
--> [378](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:378) super().__init__(nrows, ncols,
    [379](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:379)                  width_ratios=width_ratios,
...
     [50](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:50) if not isinstance(ncols, Integral) or ncols <= 0:
     [51](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:51)     raise ValueError(
     [52](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/tharvey/work/synthesizer/scripts/~/miniconda3/envs/astro/lib/python3.10/site-packages/matplotlib/gridspec.py:52)         f"Number of columns must be a positive integer, not {ncols!r}")

ValueError: Number of rows must be a positive integer, not 0
```

**Comment by WillJRoper** [comment]
Good catch @tHarvey303! We should officially hire you 😂 . Feel free to submit a fix but if not I'll fix this later today.

**Comment by WillJRoper** [comment]
I forget... but I don't think the order explicitly matters (I could be misremembering though).

**Comment by WillJRoper** [comment]
Indeed, order didn't matter. Fix implemented on #556.
